### PR TITLE
add guava in project

### PR DESCRIPTION
**Pull request by guilleSequeiraWolox**
## Summary
Add  Preconditions de Guava in setters

## Screenshots
N/A

## JIRA Card
https://trello.com/c/RMZ8KR1T/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: training/src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -83,6 +82,8 @@ public long getId() {
     }
 
     public void setId(long id) {
```
Comment (darkBasic666) [inline training/src/main/java/wolox/training/models/Book.java]: Id us autogenerate, you should't hava a setId

File: training/src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -38,19 +40,16 @@
...
-    @Column(nullable = false)
     @ApiModelProperty(notes = "year of book")
     private String year;
 
     @Column(nullable = false)
     @ApiModelProperty(notes = "pages of book")
     private Integer pages;
 
-    @Column(nullable = false)
     @ApiModelProperty(notes = "isbn of book")
```
Comment (darkBasic666) [inline training/src/main/java/wolox/training/models/Book.java:60]: Add @NotNull Annotation

**Review by darkBasic666 — CHANGES_REQUESTED** [review]

File: training/src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -147,6 +152,8 @@ public Integer getPages() {
     }
 
     public void setPages(Integer pages) {
```
Comment (rabinovichm) [inline training/src/main/java/wolox/training/models/Book.java:160]: To make it more robust, you should also check that the number of pages is greater than 0.

File: training/src/main/java/wolox/training/models/Users.java [hunk 4]
```diff
@@ -75,6 +80,8 @@ public LocalDate getBirthdate() {
     }
 
     public void setBirthdate(LocalDate birthdate) {
+        String message = "the birthdate of user must not be null";
+        Preconditions.checkNotNull(birthdate, message);
```
Comment (rabinovichm) [inline training/src/main/java/wolox/training/models/Users.java]: You could also check that the passed birthdate is previous than the actual date

**Review by rabinovichm — CHANGES_REQUESTED** [review]
Nice Job! Just two little improvements 😋 

I will also leave you [this link](https://www.baeldung.com/java-bean-validation-not-null-empty-blank) so you could quickly know the magic of @NotNull annotation and it's scope. (It's okay to use @NotNull, is not a correction, it's just a nerdy tip 🤓 )

**Review by rabinovichm — APPROVED** [review]
Nice job!

**Pull request merged**